### PR TITLE
Add top-level README, and bazel and kaniko Tasks

### PR DESCRIPTION
**Pull request by ImJasonH**
This migrates some initial contents from https://github.com/knative/build-templates

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# Tekton Task Catalog
```
Comment (dlorenc) [inline README.md]: I was thinking this would eventually contain both tasks and pipelines. WDYT?

File: kaniko/kaniko.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+apiVersion: tekton.dev/v1alpha1
```
Comment (dlorenc) [inline kaniko/kaniko.yaml]: Starting to think through the testing story here, WDYT about also including a valid TaskRun? You can inline the resources. At a minimum (you don't have to do it in this PR), we could create all of the supplied Runs and make sure they succeed.

Comment (ImJasonH) [reply kaniko/kaniko.yaml]: I had thought about it. That gets tricky when the Task requires auth to successfully push the image. It's the same reason the build-templates repo never included actual running builds... 😒

File: README.md (hunk 1 above)
Comment (ImJasonH) [reply README.md]: Yeah definitely. I only called this the "Task Catalog" because it only has Tasks for right now.

File: kaniko/kaniko.yaml (hunk 2 above)
Comment (dlorenc) [reply kaniko/kaniko.yaml]: Hmm, maybe we'll have to come up with some conventions on what auth we'll provide as part of the test. Thanks for the explanation!

File: README.md (hunk 1 above)
Comment (bobcatfish) [reply README.md]: and resources one day too (once we decouple them from the controller... WATCH THIS SPACE...)

File: README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+This repository contains a catalog of `Task` resoruces, which are designed to be
```
Comment (vdemeester) [inline README.md]: `s/resoruces/resources`

**Pull request merged**